pyfar/classes/coordinates.py

```diff
@@ -606,10 +914,264 @@ def csize(self):
+        return self._x.size
+
+    @property
+    def cartesian(self):
+        """Returns :py:func:`x`, :py:func:`y`, :py:func:`z`.
+        Right handed cartesian coordinate system. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        return np.atleast_2d(np.moveaxis(
+            np.array([self.x, self.y, self.z]), 0, -1))
+
+    @cartesian.setter
+    def cartesian(self, value):
+        self._set_points(value[..., 0], value[..., 1], value[..., 2])
+
+    @property
+    def spherical_elevation(self):
+        """Returns :py:func:`azimuth`, :py:func:`elevation`,
+        :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        azimuth, elevation, radius = cart2sph(self.x, self.y, self.z)
+        elevation = np.pi / 2 - elevation
+        return np.atleast_2d(np.moveaxis(
+            np.array([azimuth, elevation, radius]), 0, -1))
+
+    @spherical_elevation.setter
+    def spherical_elevation(self, value):
+        value[..., 1] = _check_range_angle(value[..., 1], -np.pi/2, np.pi/2)
+        x, y, z = sph2cart(
+            value[..., 0], np.pi / 2 - value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_colatitude(self):
+        """returns :py:func:`azimuth`, :py:func:`colatitude`,
+        :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        azimuth, colatitude, radius = cart2sph(self.x, self.y, self.z)
+        return np.atleast_2d(np.moveaxis(
+            np.array([azimuth, colatitude, radius]), 0, -1))
+
+    @spherical_colatitude.setter
+    def spherical_colatitude(self, value):
+        value[..., 1] = _check_range_angle(value[..., 1], 0, np.pi)
+        x, y, z = sph2cart(value[..., 0], value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_side(self):
+        """returns :py:func:`lateral`, :py:func:`polar`, :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        polar, lateral, radius = cart2sph(self.x, self.z, -self.y)
+        lateral = lateral - np.pi / 2
+        polar = np.mod(polar + np.pi / 2, 2 * np.pi) - np.pi / 2
+        return np.atleast_2d(np.moveaxis(
+            np.array([lateral, polar, radius]), 0, -1))
+
+    @spherical_side.setter
+    def spherical_side(self, value):
+        value[..., 0] = _check_range_angle(value[..., 0], -np.pi/2, np.pi/2)
+        x, z, y = sph2cart(
+            value[..., 1], np.pi / 2 - value[..., 0], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_front(self):
+        """returns :py:func:`phi`, :py:func:`theta`, :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
```

Comment:
```suggestion
        """Spherical coordinates according to the frontal pole coordinate system.
        Returns :py:func:`phi`, :py:func:`theta`, :py:func:`radius`. See
        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
        more information."""
```

Reply:
Since the documentation is the same for the setter and getter, I would suggest to also include something like 'Set or get the coordinates in the *** convention' and add a description of the input parameters that could be copied from the `from_*` methods.